server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -931,7 +883,7 @@ public TSFetchResultsResp fetchResults(TSFetchResultsReq req) {
       } else {
         result = QueryDataSetUtils.convertQueryDataSetByFetchSize(queryDataSet, req.fetchSize);
       }
-      boolean hasResultSet = (result.getRowCount() != 0);
+      boolean hasResultSet = (result.bufferForTime().limit() != 0);
```

Comment:
limit or remaining?

Reply:
Both ok, when there is no data, limit=position=0